aawordsearch.c

```diff
@@ -382,18 +389,24 @@ print_answer_key (FILE * restrict stream, char puzzle[][GRID_SIZE])
 
 
 void
-print_puzzle (FILE * restrict stream, char puzzle[][GRID_SIZE])
+print_puzzle (FILE * restrict stream, wchar_t puzzle[][GRID_SIZE])
 {
+  wchar_t *alphabet = NULL;
+  if (strcmp (lang, "en") == 0)
+    alphabet = en_alphabet;
+  else if (strcmp (lang, "de") == 0)
+    alphabet = de_alphabet;
+
   int i, j;
   for (i = 0; i < GRID_SIZE; i++)
   {
     for (j = 0; j < GRID_SIZE; j++)
     {
       if (puzzle[i][j] == fill_char)
-        fprintf (stream, "%c ",
-                 (rand () % ((int) 'Z' - (int) 'A' + 1)) + (int) 'A');
+        fprintf (stream, "%lc ",
+                 alphabet[rand () % sizeof(alphabet) / sizeof(alphabet[0])]);
```

Comment:
Something isn't right here, @Jammyjamjamman . Note how a lot of letters (especially 'A' are repeated in the output.

Reply:
Because I was getting the size of a pointer... Almost fixed with the last commit